diff -r 8e0661d279a9 configure.in
--- configure.in	Fri Nov 27 11:26:34 2009 -0800
+++ configure.in	Fri Dec 18 14:42:44 2009 -0800
@@ -4263,10 +4263,14 @@ else
         MOZ_PYTHON_LIBS="-L$MOZ_PYTHON_PREFIX -lpython$MOZ_PYTHON_VER_DOTTED"
         AC_MSG_ERROR([The Python at $MOZ_PYTHON_PREFIX looks like a dev tree. The PyXPCOM build cannot handle this yet. You must 'make install' Python and use the installed tree.])
     elif test "$OS_ARCH" = "Darwin"; then
-        dnl We do Darwin last, so if a custom non-framework build of
-        dnl python is used on OSX, then it will be picked up first by
-        dnl the logic above.
-        MOZ_PYTHON_LIBS="-framework Python"
+        dnl Mac needs to link against the provided Python, not system Python
+        dnl .../Python.framework/Versions/2.6/bin/python
+        MOZ_PYTHON_FRAMEWORK=`dirname "$MOZ_PYTHON"`
+        MOZ_PYTHON_FRAMEWORK=`dirname "$MOZ_PYTHON_FRAMEWORK"`
+        MOZ_PYTHON_FRAMEWORK=`dirname "$MOZ_PYTHON_FRAMEWORK"`
+        MOZ_PYTHON_FRAMEWORK=`dirname "$MOZ_PYTHON_FRAMEWORK"`
+        MOZ_PYTHON_FRAMEWORK=`dirname "$MOZ_PYTHON_FRAMEWORK"`
+        MOZ_PYTHON_LIBS="-F$MOZ_PYTHON_FRAMEWORK -framework Python"
     else
         AC_MSG_ERROR([Could not find build shared libraries for Python at $MOZ_PYTHON_PREFIX.  This is required for PyXPCOM.])
     fi
